File: 2021/A/K_goodness.py
T = int(input())

def score(S, N):
    goodness = 0
    for i in range(1, int(N/2) + 1):
        if (S[i-1] != S[N-1-i+1]):
            goodness += 1
                
            
    return goodness

for i in range(T):
    N, K = [int(s) for s in input().split(" ")]
    
    S = str(input())
    
    goodness = score(S, N)
    
    if (goodness == K):
        print("Case #" + str(i+1) + ": 0")
    else:
        print("Case #" + str(i+1) + ": " + str(abs(K-goodness)))
        

File: 2021/A/K_goodness.cpp
#include <iostream>
#include <cmath>
using namespace std;


int score(string s, int k) {
    int x = 0;
    for (int i =0; i < s.size() / 2;i++) {
        if (s[i] != s[s.size() - 1- i])
            x++;
    }


    return abs(x-k);

}

int main() {
    int t,n,k;
    string s;

    cin >> t;

    for (int i = 0; i<t;i++) {
        cin >> n >> k >> s;

        int y = score(s, k);
        cout << "Case #" << i + 1 << ": " << y << "\n";
    }

    return 0;
}